@xen-orchestra/backups/disks/RemoteVhdDisk.mjs

```diff
@@ -0,0 +1,239 @@
+                index.push(blockIndex)
+            }
+        }
+        return index
+    }
+
+    /**
+     * Writes a full block into this VHD.
+     * @param {number} index
+     * @param {Buffer} data
+     * @return {number}
+     */
+    async writeBlock(index, data) {
+        if (this.#vhd === undefined) {
+            throw new Error(`can't call readBlock of a RemoteVhdDisk before init`)
+        }
+        await this.#vhd.writeEntireBlock({ id: index, buffer: data })
+        await this.#vhd.writeBlockAllocationTable()
+
+        return this.getBlockSize();
+    }
+
+    /**
+     * Reads a specific block from the VHD.
+     * @param {number} index
+     * @returns {Promise<DiskBlock>}
+     */
+    async readBlock(index) {
+        if (this.#vhd === undefined) {
+            throw new Error(`can't call readBlock of a RemoteVhdDisk before init`)
+        }
+        const { data } = await this.#vhd.readBlock(index)
+        return {
+            index,
+            data,
+        }
+    }
+
+    /**
+     * @returns {DiskMetadata}
+     */
+    getMetadata() {
+        if (this.#vhd === undefined) {
+            throw new Error(`can't call setvirtualsize of a RemoteVhdDisk before init`)
+        }
+        
+        return this.#vhd.footer;
+    }
+
+    /**
+     * @param {RemoteDisk} child
+     */
+    mergeMetadata(child) {
+        const childMetadata = child.getMetadata();
+
+        if (this.#vhd === undefined) {
+            throw new Error(`can't call mergeMetadata of a RemoteVhdDisk before init`)
+        }
+
+        this.#vhd.footer.currentSize = childMetadata.currentSize
+        this.#vhd.footer.diskGeometry = { ...childMetadata.diskGeometry }
+        this.#vhd.footer.originalSize = childMetadata.originalSize
+        this.#vhd.footer.timestamp = childMetadata.timestamp
+        this.#vhd.footer.uuid = childMetadata.uuid
+    }
+
+    /**
+     * @param {DiskMetadata} metadata
+     */
+    setMetadata(metadata) {
+        if (this.#vhd === undefined) {
+            throw new Error(`can't call setvirtualsize of a RemoteVhdDisk before init`)
+        }
+
+        this.#vhd.footer.currentSize = metadata.currentSize ?? this.#vhd.footer.currentSize
+        this.#vhd.footer.diskGeometry = metadata.diskGeometry ?? this.#vhd.footer.diskGeometry
+        this.#vhd.footer.originalSize = metadata.originalSize ?? this.#vhd.footer.originalSize
+        this.#vhd.footer.timestamp = metadata.timestamp ?? this.#vhd.footer.timestamp
+        this.#vhd.footer.uuid = metadata.uuid ?? this.#vhd.footer.uuid
+    }
+
+    /**
+     * Writes block allocation table
+     */
+    async writeBlockAllocationTable() {
+        if (this.#vhd === undefined) {
+            throw new Error(`can't call readBlock of a RemoteVhdDisk before init`)
+        }
+        await this.#vhd.writeBlockAllocationTable()
+    }
+
+    /**
+     * Deletes disk
+     */
+    async unlink() {
+        await this.close()
+        await this.#handler.unlink(this.#path)
```

Comment:
it should call Vhd.unlink instead, since a vhd can be a directory or have an alias